### PR TITLE
Android: update Gradle, build tools and Apache Commons IO dependency to the latest version

### PR DESCRIPTION
**Pull request by oleg-derevenetz**

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -7,7 +7,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.0.3'
+        classpath 'com.android.tools.build:gradle:7.3.1'
```
Comment (ihhub) [inline android/build.gradle:10]: Here we specify 7.3.1 version while we download 7.4. Is it intentional?

File: android/app/build.gradle [hunk 2]
```diff
@@ -40,7 +42,7 @@ android {
 
 dependencies {
     implementation project(':sdl2')
-    implementation 'org.apache.directory.studio:org.apache.commons.io:2.4'
+    implementation group: 'commons-io', name: 'commons-io', version: '2.11.0'
```
Comment (ihhub) [inline android/app/build.gradle:45]: Could you please explain where this magic number 2.11.0 belongs to?

**Review by ihhub — COMMENTED** [review]
Hi @oleg-derevenetz , I left 2 questions in this pull request. Could you please check them?

File: android/build.gradle (hunk 1 above)
Comment (oleg-derevenetz) [reply android/build.gradle:10]: Hi @ihhub These are different things. This number is related to the version of Android build tools package from Google. 7.4 is related to the version of the Gradle itself.

File: android/app/build.gradle (hunk 2 above)
Comment (oleg-derevenetz) [reply android/app/build.gradle:45]: Hi @ihhub it is related to the version of the corresponding build artifact in the Maven repository:

https://mvnrepository.com/artifact/commons-io/commons-io

File: android/build.gradle (hunk 1 above)
Comment (ihhub) [reply android/build.gradle:10]: Thank you for the explanation!

File: android/app/build.gradle (hunk 2 above)
Comment (ihhub) [reply android/app/build.gradle:45]: Thank you for the explanation!

**Comment by ihhub** [comment]
@oleg-derevenetz , thank you very much for this improvement!

File: android/build.gradle (hunk 1 above)
Comment (oleg-derevenetz) [reply android/build.gradle:10]: @ihhub More precisely this is called "Android plugin":

https://developer.android.com/studio/releases/gradle-plugin#7-3-0

:)

**Pull request merged**